### ISSUE TITLE
iOS 11 and greater: code should refer locale id.

### ISSUE THREAD

**Issue by phackwer** [magnatronus/flutter-devicelocale#53]
Due iOS 11 and greater limitation, app is not getting the locale from the device, and in some cases it's always returning en-US.

The problem is that the implementation should be using the locale id instead, otherwise it will always return what has been compiled instead of the real device locale.

Please, refer those docs and the comment on stackoverflow

https://developer.apple.com/forums/thread/88711

https://stackoverflow.com/questions/46205763/nsdateformatter-and-current-language-in-ios11/46206511#46206511

https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/LanguageandLocaleIDs/LanguageandLocaleIDs.html

**Comment by phackwer** [comment]
The change I mention about is basically just replace line 17 by those 2
```
    NSString *localeID = NSLocale.preferredLanguages.firstObject;
    NSString *locale = [[NSLocale localeWithLocaleIdentifier:localeID] objectForKey: NSLocaleCountryCode];
```

**Comment by edwardmp** [comment]
Possible duplicate of #38

**Comment by magnatronus** [comment]
closed - dupe of [ #38](https://github.com/magnatronus/flutter-devicelocale/issues/38)
